src/java.desktop/unix/native/common/awt/fontpath.c

```diff
@@ -1100,7 +1105,11 @@ Java_sun_font_FontConfigManager_getFontConfig
             } else {
                 if ((*FcCharSetSubtractCount)(charset, unionCharset)
                     > minGlyphs) {
-                    unionCharset = (* FcCharSetUnion)(unionCharset, charset);
+                    FcCharSet *currentUnionSet = unionCharset;
+                    unionCharset = (* FcCharSetUnion)(currentUnionSet, charset);
+                    if (currentUnionSet != charset) {
+                        (*FcCharSetDestroy)(currentUnionSet);
+                    }
```

Comment:
Hmm. I worry that you may be replacing a leak with a crash.
The original "charset" was returned from FcPatternGetCharSet() and I don't think it was a copy and will be freed when the pattern is destroyed.
So that should not be freed here.

Reply:
Right, that is why I have `if (currentUnionSet != charset)` check to prevent from freeing original `charset`.